lib/ip_wrangler/db.rb

```diff
@@ -0,0 +1,85 @@
+module IpWrangler
+  class DB
+    def initialize(db_name, logger)
+      @db = Sequel.connect('sqlite://' + db_name)
+      @logger = logger
+    end
+
+    def select_nat_port(private_ip=nil, private_port=nil, protocol=nil)
+      params = {:private_ip => private_ip, :private_port => private_port, :protocol => protocol}.select { |key, value| value != nil }
+      nat_ports = []
+      @db[:nat_ports].where(params).each do |nat_port|
+        if nat_port[:private_port] != nil and nat_port[:private_port] != nil
+          nat_ports.push nat_port
+        end
+      end
+      nat_ports
+    end
+
+    def select_nat_ip(private_ip=nil, public_ip=nil)
+      params = {:private_ip => private_ip, :public_ip => public_ip}.select { |key, value| value != nil }
+      nat_ips = []
+      @db[:nat_ips].where(params).each do |nat_ip|
+        if nat_ip[:private_ip] != nil
+          nat_ips.push nat_ip
+        end
+      end
+      nat_ips
+    end
+
+    def insert_nat_port(public_ip, public_port, private_ip, private_port, protocol)
+      params = {:public_ip => public_ip, :public_port => public_port, :protocol => protocol}
```

Comment:
Line is too long. [92/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.